### PR TITLE
Implement Sankey charts for NVD3

### PR DESCRIPTION
**Pull request by ksgy**
This PR is a simple implementation of the Sankey Diagram by 
- https://bost.ocks.org/mike/sankey/
- https://github.com/soxofaan/d3-plugin-captain-sankey

The PR has:
- Example page with a simple and an advanced version of Sankey chart
- Basic `mocha` tests for `svg` structure and attributes
- Sankey model
- Sankey chart with error handling for "no data" and data format check

### CODE REVIEW

File: src/models/sankeyChart.js [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    };
+
+
+    //============================================================
+
+    function chart(selection) {
+        selection.each(function(data) {
+
+            //============================================================
+            // Error handling
```
Comment (liquidpele) [inline src/models/sankeyChart.js]: What's with these random comments that don't belong?     Please just remove any unhelpful comments, including just random lines of equal signs :)

File: src/models/sankey.js [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+        size = [1, 1],
+        nodes = [],
+        links = [],
+        sinksRight = true;
+
+    sankey.nodeWidth = function(_) {
+        if (!arguments.length) return nodeWidth;
+        nodeWidth = +_;
+        return sankey;
+    };
```
Comment (liquidpele) [inline src/models/sankey.js]: It looks like the options on this file aren't using the chart._options format, which makes it so they can't be inherited automatically, please make this fit the same format as the rest of nvd3.

File: src/models/sankeyChart.js (hunk 1 above)
Comment (ksgy) [inline src/models/sankeyChart.js]: Seen in https://github.com/novus/nvd3/blob/master/src/models/line.js - thought make the code more readable, but will remove :)

File: src/models/sankey.js (hunk 2 above)
Comment (ksgy) [inline src/models/sankey.js]: Missed that one, will do that

File: src/models/sankeyChart.js (hunk 1 above)
Comment (liquidpele) [inline src/models/sankeyChart.js]: I tried to remove them all, guess I missed some  :p 

originally there were like 20 of those per file with no real reasoning as to where they were, so it actually impeded readability.

File: src/models/sankey.js (hunk 2 above)
Comment (ksgy) [inline src/models/sankey.js]: @liquidpele Refactored using the `_options`but not sure if that's the right way to pass and call functions, I haven't seen any kind of similar "callback" function to start the rendering for a chart.
See the last 3 properties: https://github.com/ksgy/nvd3/blob/master/src/models/sankey.js#L309-L311

**Comment by ksgy** [comment]
Hey @liquidpele any news on this pull request?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "nvd3",
-  "version": "1.8.4-dev",
+  "version": "1.8.5-dev",
```
Comment (liquidpele) [inline package.json]: Wait... why is this changing the version?

Comment (ksgy) [inline package.json]: I might be wrong, but as far as I remember, all previous new features increased the version, but I can change it back :)

Comment (liquidpele) [inline package.json]: Nope, please change it back.

**Pull request merged**